### DISCUSSION TITLE
Bundler: How to run `bundle` command from repo on Bundler 1.

### DISCUSSION THREAD

**Discussion by junaruga** [ruby/rubygems#5115] [Q&A] [answered]
I was told the way to run the `bundle` command from repository is like this on the latest Bundler and Bundler 2.

```
$ ruby bundler/spec/support/bundle.rb --version
Bundler version 2.3.0.dev
```

But do you remember how did you run the `bundle` command from repository on Bundler 1? On Bundler 1, there is no `spec/support/bundle.rb`. For example how do you run it on the `bundler-v1.17.3` branch? It's useful to do my work on the downstream Bundler 1, if I know it.

Thanks.

**Comment by deivid-rodriguez** [comment, marked as answer]
I don't remember :( You could run `exe/bundle` directly, the only difference is that `bundler` is not activated as a gem (that's what `bundler/spec/support/bundle.rb` does). Not sure whether that has consequences or not for you. Also, not sure how `exe/bundle` required things at the time. You might also need to pass `-Ibundler/lib` to ruby.

**Reply by junaruga** [reply]
Thanks for the advice! I succeeded to run the `bundle` command from repository on the `bundler-v1.17.3` branch now. Here is the log. On the newer Ruby bundling Bundler, I needed to specify the `-I lib` to load the intended Bundler by `require "bundler"` in the `exe/bundle`. The `-I lib` is same with `-Ibundler/lib` on the Bundler 2, the directory structure was different for the `bundler-v1.17.3` branch, 

```
$ which ruby
/usr/local/ruby-3.0.2/bin/ruby

$ ruby -v
ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux]

$ which bundle
/usr/local/ruby-3.0.2/bin/bundle

$ bundle -v
Bundler version 2.2.22
```

The `rubygems` git directory is on the `bundler-v1.17.3` branch. But without `ruby -I` option, the default installed Bundler 2.2.22 is loaded.

```
$ ~/git/rubygems/exe/bundle --version
Bundler version 2.2.22
```

With the `ruby -I <bundler lib directory>`, the intended Bundler is loaded. 

```
$ ruby -I ~/git/rubygems/lib -S ~/git/rubygems/exe/bundle --version
/home/jaruga/var/git/rubygems/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
/home/jaruga/var/git/rubygems/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
Bundler version 1.17.3
```

**Reply by junaruga** [reply]
I also succeeded to run the `bundle` command from the Ruby 2.6.8 source repository. 👍🏼  My work is much efficient now. 😄 

```
$ which ruby
/usr/bin/ruby

$ ruby -v
ruby 2.6.8p205 (2021-07-07 revision 67951) [x86_64-linux]
```

The Bundler is not installed on this environment.

```
$ which bundle
/usr/bin/which: no bundle in (/builddir/.local/bin:/builddir/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin)
```

Run the `bundle` command from the repository like this. The `/builddir/build/BUILD/ruby-2.6.8` is Ruby's source directory.

```
$ ruby -I /builddir/build/BUILD/ruby-2.6.8/lib -S /builddir/build/BUILD/ruby-2.6.8/libexec/bundle --version
Bundler version 1.17.2
```

When running with `byebug`, the command is like this.

```
RUBYOPT="-I /builddir/build/BUILD/ruby-2.6.8/lib" byebug ruby /builddir/build/BUILD/ruby-2.6.8/libexec/bundle --version
```
